examples/ateccx08/random/rand.go

```diff
@@ -0,0 +1,52 @@
+// connects the Go crypto/rand package to the random number generation
+// on the ATECCx08 cryptographic processor.
+package main
+
+import (
+	"crypto/rand"
+	"errors"
+)
+
+var (
+	errNoATECC = errors.New("no ATECCx08")
+)
+
+func init() {
+	rand.Reader = &reader{}
```

Comment:
I'm all for putting this in the main package, but I think it would be great to create a `Reader` object in the driver itself, so you could do something like this:

```go
func main() {
    // ...
    rand.Reader = ateccx08.NewRand(atecc) // or a different name, NewRandReader?
    // ...
}
```